parameterspace/configspace_utils.py

```diff
@@ -1,20 +1,58 @@
 """Initialize a `ParameterSpace` from a `ConfigSpace` JSON dictionary."""
 
-from typing import List, Optional, Tuple
+from typing import Dict, List, Optional, Tuple
 
 import numpy as np
 
 import parameterspace as ps
 from parameterspace.condition import Condition
 from parameterspace.utils import verify_lambda
 
+RESERVED_WORDS = [
+    "and",
+    "as",
+    "assert",
+    "break",
+    "class",
+    "continue",
+    "def",
+    "del",
+    "elif",
+    "else",
+    "except",
+    "finally",
+    "for",
+    "form",
+    "global",
+    "if",
+    "import",
+    "in",
+    "is",
+    "lambda",
+    "nonlocal",
+    "not",
+    "or",
+    "pass",
+    "raise",
+    "return",
+    "try",
+    "while",
+    "with",
+    "yield",
+]
 
-def _escape_parameter_name(name: str) -> str:
-    """Replace colons with underscores.
 
-    Colons are incompatible as ParameterSpace parameter names.
+def _escape_parameter_name(name: str) -> str:
+    """Replace colons, dashes, dots and spaces with an underscore and add an underscore
+    suffix to reserved Python words.
     """
-    return name.replace(":", "_")
+    name = name.replace(":", "_").replace("-", "_").replace(".", "_").replace(" ", "_")
+
+    for reserved_word in RESERVED_WORDS:
```

Comment:
Can we not use the [keyword](https://docs.python.org/3.9/library/keyword.html#module-keyword) standard library? The function `keyword.iskeyword` looks exactly like the functionality we try to replicate here.

Reply:
If only we had thought of this before :smile: :see_no_evil: 
https://github.com/boschresearch/parameterspace/blob/f8fdc79b93e9b31e74e6d577ee61ff496971c9af/parameterspace/parameters/base.py#L18